framework/src/outputs/CSV.C

```diff
@@ -90,6 +90,21 @@ CSV::initialSetup()
       MooseUtils::clearSymlink(out_final);
     }
   }
+
+  // See https://github.com/idaholab/moose/issues/25211.
+  const auto pp_execute_on = advancedExecuteOn().find("postprocessors")->second;
+  const auto scalar_execute_on = advancedExecuteOn().find("scalars")->second;
+  const auto reporter_execute_on = advancedExecuteOn().find("reporters")->second;
```

Comment:
Use `libmesh_map_find` for these, so we at least die horribly if we can't find these

Reply:
Am I doing something wrong?
```
/Users/hansje/projects/sockeye/moose/framework/src/outputs/CSV.C:95:30: error: no matching function for call to 'map_find'
  const auto pp_execute_on = libmesh_map_find(advancedExecuteOn(), "postprocessors");
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:39:36: note: expanded from macro 'libmesh_map_find'
#define libmesh_map_find(map, key) libMesh::Utility::map_find((map), (key), __FILE__, __LINE__)
                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:89:1: note: candidate template ignored: requirement '!is_streamable<char [15]>::value' was not satisfied [with Map = const OutputOnWarehouse, Key = char [15]]
map_find(Map & map,
^
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:108:1: note: candidate template ignored: requirement '!is_streamable<char [15]>::value' was not satisfied [with Map = OutputOnWarehouse, Key = char [15]]
map_find(const Map & map,
^
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:128:1: note: candidate template ignored: substitution failure [with Map = const OutputOnWarehouse, Key = char [15], $2 = nullptr]: no type named 'mapped_type' in 'OutputOnWarehouse'
map_find(Map & map,
^
/Users/hansje/projects/sockeye/moose/scripts/../libmesh/installed/include/libmesh/utility.h:147:1: note: candidate template ignored: substitution failure [with Map = OutputOnWarehouse, Key = char [15], $2 = nullptr]: no type named 'mapped_type' in 'OutputOnWarehouse'
map_find(const Map & map,
```
Does `libmesh_map_find` work with this wrapped map class `OutputMapWrapper`?

Reply:
Oops. It doesn't. Just add the following above then:

```
mooseAssert(advancedExecuteOn().contains("postprocessors"), "Missing expected postprocessors key");
mooseAssert(advancedExecuteOn().contains("scalars"), "Missing expected scalars key");
mooseAssert(advancedExecuteOn().contains("reporters"), "Missing expected reporters key");
```